### PR TITLE
feat: Add retry delay as span attribute in retry handler

### PR DESCRIPTION
**Pull request by ohad83**

### CODE REVIEW

File: retry_handler.go [hunk 1]
```diff
@@ -144,6 +144,7 @@ func (middleware RetryHandler) retryRequest(ctx context.Context, pipeline Pipeli
 			ctx, span := otel.GetTracerProvider().Tracer(observabilityName).Start(ctx, "RetryHandler_Intercept - attempt "+fmt.Sprint(executionCount))
 			span.SetAttributes(attribute.Int("http.retry_count", executionCount),
```
Comment (baywet) [inline retry_handler.go:145]:
```suggestion
			span.SetAttributes(attribute.Int("http.request.resend_count", executionCount),

```

https://opentelemetry.io/docs/specs/semconv/http/http-spans/

File: retry_handler.go [hunk 2]
```diff
@@ -144,6 +144,7 @@ func (middleware RetryHandler) retryRequest(ctx context.Context, pipeline Pipeli
 			ctx, span := otel.GetTracerProvider().Tracer(observabilityName).Start(ctx, "RetryHandler_Intercept - attempt "+fmt.Sprint(executionCount))
 			span.SetAttributes(attribute.Int("http.retry_count", executionCount),
 				attribute.Int("http.status_code", resp.StatusCode),
+				attribute.Float64("http.retry_delay", delay.Seconds()),
```
Comment (baywet) [inline retry_handler.go:147]:
```suggestion
				attribute.Float64("http.request.resend_delay", delay.Seconds()),

```

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for the contribution!
Here are a couple of suggestions.
In addition to this, can you also please:
- bump the version in the user agent options (patch)
- add an entry in the changelog entry (today's date, patch version, changed)

**Comment by ohad83** [comment]
@microsoft-github-policy-service agree company="MIND"

**Comment by ohad83** [comment]
Thanks for the review! Added the requested changes.

**Pull request merged**